packages/main/src/plugin/contribution-manager.ts

```diff
@@ -168,4 +199,144 @@ export class ContributionManager {
   listContributions(): ContributionInfo[] {
     return this.contributions;
   }
+
+  async enhanceComposeFile(
+    rootDirectory: string,
+    ociImageName: string,
+    metadata: DockerExtensionMetadata,
+  ): Promise<void> {
+    // if there is a VM file, generate it
+    if (metadata?.vm?.composefile) {
+      // read this file
+      const composeFile = path.join(rootDirectory, metadata.vm.composefile);
+
+      if (!fs.existsSync(composeFile)) {
+        throw new Error(`The file ${composeFile} does not exists.`);
+      }
+
+      const content = await fs.promises.readFile(composeFile, 'utf-8');
+
+      // parse it as yaml file
+      const composeObject = jsYaml.load(content) as ComposeObject;
+
+      // read port number from the global port file
+      const globalPortsFile = path.join(rootDirectory, '..', ContributionManager.GLOBAL_PORTS_FILE);
+
+      let initPortRange = 10000;
+      // if does not exist, create it with 10000
+      if (fs.existsSync(globalPortsFile)) {
+        const portNumberString = await fs.promises.readFile(globalPortsFile, 'utf-8');
+        initPortRange = parseInt(portNumberString) + 1;
```

Comment:
shouldn't this be inside a try/catch? It may fail with a misconfigured file

Reply:
I've added a check on isNan + a unit test